modules/react/breadcrumbs/lib/BreadcrumbsOverflowButton.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import {chevronRightSmallIcon, relatedActionsIcon} from '@workday/canvas-system-icons-web';
+import {colors, space} from '@workday/canvas-kit-react/tokens';
+import {SystemIcon} from '@workday/canvas-kit-react/icon';
+import {Flex, FlexProps} from '@workday/canvas-kit-react/layout';
+import {
+  createElemPropsHook,
+  composeHooks,
+  subModelHook,
+  createSubcomponent,
+  useIsRTL,
+} from '@workday/canvas-kit-react/common';
+import {useOverflowListTarget} from '@workday/canvas-kit-react/collection';
+import {useMenuTarget} from '@workday/canvas-kit-react/menu';
+import {useBreadcrumbsModel} from './hooks/useBreadcrumbsModel';
+import {TertiaryButton, TertiaryButtonProps} from '@workday/canvas-kit-react/button';
+
+export const useBreadcrumbsOverflowButton = composeHooks(
+  createElemPropsHook(useBreadcrumbsModel)((_model, _element) => ({
+    'aria-haspopup': true,
+    'aria-controls': 'menu',
+  })),
+  useOverflowListTarget,
+  subModelHook((m: ReturnType<typeof useBreadcrumbsModel>) => m.menu, useMenuTarget)
+);
+
+export const BreadcrumbsOverflowButton = createSubcomponent('button')({
+  displayName: 'Breadcrumbs.OverflowButton',
+  modelHook: useBreadcrumbsModel,
+  elemPropsHook: useBreadcrumbsOverflowButton,
+})<TertiaryButtonProps>((elemProps, Element) => {
+  const styles = (elemProps as typeof elemProps & {style: FlexProps}).style;
```

Comment:
this is kinda neat!

Reply:
I like this too. 😄 

Q: I'm not sure I understand how `.style` is always an object though? Maybe I'm misreading the syntax, but it seems like if `elemProps` doesn't include a `style` key, it would return `undefined`, and because we're casting it with `as` TS wouldn't catch that. What am I misunderstanding?

Reply:
Most of the time, the `OverflowButton` will be implicitly rendered instead of a developer implementing it, so it often won't matter. But in the case they do, I'd like to make sure they can't mess it up unintentionally.